src/backend/types.js

```diff
@@ -7,8 +7,88 @@ type BundleType =
   | 0 // PROD
   | 1; // DEV
 
-// TODO: Better type for Fiber
-export type Fiber = Object;
+export type WorkTag = number;
+export type TypeOfMode = number;
+export type SideEffectTag = number;
+export type ExpirationTime = number;
+export type RefObject = {|
+  current: any,
+|};
+export type Source = {
+  fileName: string,
+  lineNumber: number,
+};
+export type HookType =
+  | 'useState'
+  | 'useReducer'
+  | 'useContext'
+  | 'useRef'
+  | 'useEffect'
+  | 'useLayoutEffect'
+  | 'useCallback'
+  | 'useMemo'
+  | 'useImperativeHandle'
+  | 'useDebugValue';
+
+// Copied modified from https://github.com/facebook/react/blob/v16.9.0-alpha.0/packages/react-reconciler/src/ReactFiber.js
+export type Fiber = {|
+  tag: WorkTag,
+
+  key: null | string,
+
+  elementType: any,
+
+  type: any,
+
+  stateNode: any,
+
+  return: Fiber | null,
+
+  child: Fiber | null,
+  sibling: Fiber | null,
+  index: number,
+
+  ref: null | (((handle: mixed) => void) & { _stringRef: ?string }) | RefObject,
+
+  pendingProps: any, // This type will be more specific once we overload the tag.
+  memoizedProps: any, // The props used to create the output.
+
+  // updateQueue: UpdateQueue<any> | null,
+
+  memoizedState: any,
+
+  // contextDependencies: ContextDependencyList | null,
+
+  mode: TypeOfMode,
+
+  effectTag: SideEffectTag,
+
+  nextEffect: Fiber | null,
+
+  firstEffect: Fiber | null,
+  lastEffect: Fiber | null,
+
+  expirationTime: ExpirationTime,
+
+  childExpirationTime: ExpirationTime,
```

Comment:
We should omit the properties we don't use in DevTools.

Reply:
> We should omit the properties we don't use in DevTools.

👉 https://github.com/bvaughn/react-devtools-experimental/pull/240/commits/d35cf80921245ff7da4bcc7c17f68082f21088fb